tests/sanity/sanity_test.go

```diff
@@ -0,0 +1,356 @@
+					return nil, cloud.ErrAlreadyExists
+				}
+			}
+
+			if diskOptions.SnapshotID != "" {
+				if _, exists := snapshots[diskOptions.SnapshotID]; !exists {
+					return nil, cloud.ErrNotFound
+				}
+			}
+
+			newDisk := &cloud.Disk{
+				VolumeID:         volumeID,
+				AvailabilityZone: diskOptions.AvailabilityZone,
+				CapacityGiB:      util.BytesToGiB(diskOptions.CapacityBytes),
+			}
+			disks[volumeID] = newDisk
+			return newDisk, nil
+		},
+	).AnyTimes()
+
+	// DeleteDisk
+	c.EXPECT().DeleteDisk(gomock.Any(), gomock.Any()).DoAndReturn(
+		func(ctx context.Context, volumeID string) (bool, error) {
+			_, exists := disks[volumeID]
+			if !exists {
+				return false, cloud.ErrNotFound
+			}
+			delete(disks, volumeID)
+			return true, nil
+		},
+	).AnyTimes()
+
+	// GetDiskByID
+	c.EXPECT().GetDiskByID(gomock.Any(), gomock.Any()).DoAndReturn(
+		func(ctx context.Context, volumeID string) (*cloud.Disk, error) {
+			disk, exists := disks[volumeID]
+			if !exists {
+				return nil, cloud.ErrNotFound
+			}
+			return disk, nil
+		},
+	).AnyTimes()
+
+	// CreateSnapshot
+	c.EXPECT().
+		CreateSnapshot(gomock.Any(), gomock.Any(), gomock.Any()).DoAndReturn(
+		func(ctx context.Context, volumeID string, opts *cloud.SnapshotOptions) (*cloud.Snapshot, error) {
+			snapshotID := fmt.Sprintf("snapshot-%d", rand.New(rand.NewSource(time.Now().UnixNano())).Uint64())
+
+			_, exists := snapshots[snapshotID]
+			if exists {
+				return nil, cloud.ErrAlreadyExists
+			}
+			newSnapshot := &cloud.Snapshot{
+				SnapshotID:     snapshotID,
+				SourceVolumeID: volumeID,
+				CreationTime:   time.Now(),
+			}
+			snapshots[snapshotID] = newSnapshot
+			snapshotNameToID[opts.Tags["CSIVolumeSnapshotName"]] = snapshotID
+			return newSnapshot, nil
+		}).AnyTimes()
+
+	// DeleteSnapshot
+	c.EXPECT().DeleteSnapshot(gomock.Any(), gomock.Any()).DoAndReturn(
+		func(ctx context.Context, snapshotID string) (bool, error) {
+			if _, exists := snapshots[snapshotID]; !exists {
+				return false, cloud.ErrNotFound
+			}
+			for name, id := range snapshotNameToID {
+				if id == snapshotID {
+					delete(snapshotNameToID, name)
+					break
+				}
+			}
+			delete(snapshots, snapshotID)
+			return true, nil
+		},
+	).AnyTimes()
+
+	// GetSnapshotByID
+	c.EXPECT().GetSnapshotByID(gomock.Any(), gomock.Any()).DoAndReturn(
+		func(ctx context.Context, snapshotID string) (*cloud.Snapshot, error) {
+			snapshot, exists := snapshots[snapshotID]
+			if !exists {
+				return nil, cloud.ErrNotFound
+			}
+			return snapshot, nil
+		},
+	).AnyTimes()
+
+	// GetSnapshotByName
+	c.EXPECT().GetSnapshotByName(gomock.Any(), gomock.Any()).DoAndReturn(
+		func(ctx context.Context, name string) (*cloud.Snapshot, error) {
+			if snapshotID, exists := snapshotNameToID[name]; exists {
+				return snapshots[snapshotID], nil
+			}
+			return nil, cloud.ErrNotFound
+		},
+	).AnyTimes()
+
+	// ListSnapshots
+	c.EXPECT().
+		ListSnapshots(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).
```

Comment:
[p](panic: reflect: Call using int32 as type int64) appear to be resulting from here